internal/models/integration.go

```diff
@@ -38,6 +39,9 @@ type KubernetesSecret struct {
 
 func ConvertToIntegrationModel(ctx context.Context, integration *apono.Integration) (*IntegrationModel, diag.Diagnostics) {
 	metadataMapValue, diagnostics := types.MapValueFrom(ctx, types.StringType, integration.GetMetadata())
+	if len(diagnostics) > 0 {
```

Comment:
lets check what is the correct way of using those diagnostics

Reply:
checked and it seems to be one of the correct ways to use diagnostics.